app/src/main/java/com/quark/client/database/Users.kt

```diff
@@ -73,6 +77,51 @@ class Users(
                 continuation.resumeWithException(exception)
             }
     }
+
+    /**
+     * Takes a prompted username, then searches firebase to see if it exists, Used as a dependency
+     * for setUsername
+     * @param username: String
+     * @return Boolean
+     * @see UserProfile
+     */
+
+    private fun usernameUsedQuery(username: String): Boolean {
```

Comment:
needs to be suspend with coroutine. currently, this will always return true

Reply:
i think

Reply:
Done